### PR TITLE
Get FileHandle working on NodeJS

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Review by oldergod — APPROVED** [review]

File: okio-nodefilesystem/src/jsMain/kotlin/okio/NodeJsFileHandle.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  ): Int {
+    val readByteCount = readSync(
+      fd = fd,
+      buffer = array,
+      length = byteCount,
+      offset = arrayOffset,
+      position = fileOffset.toDouble()
+    ).toInt()
+
+    if (readByteCount == 0) return -1
```
Comment (oldergod) [inline okio-nodefilesystem/src/jsMain/kotlin/okio/NodeJsFileHandle.kt:42]: So we can never return zero. Is that a standard spec?

**Review by oldergod — APPROVED** [review]

File: okio-nodefilesystem/src/jsMain/kotlin/okio/NodeJsFileHandle.kt (hunk 1 above)
Comment (swankjesse) [reply okio-nodefilesystem/src/jsMain/kotlin/okio/NodeJsFileHandle.kt:42]: exactly!

**Comment by swankjesse** [comment]
Arg, it’s broken only on JS + Windows. Windows works, JS works, but the combination is bad.

```

> Task :okio:jsLegacyNodeTest

okio.NodeJsFileSystemTest.fileHandleWriteAndOverwrite FAILED
    AssertionError at D:\a\okio\okio\build\js\node_modules\kotlin\kotlin\builtins.kt:103

okio.NodeJsFileSystemTest.fileHandleWriteBeyondEnd FAILED
    EOFException at D:\a\okio\okio\build\js\node_modules\kotlin\kotlin\builtins.kt:103

okio.NodeJsFileSystemTest.fileHandleResizeSmaller FAILED
    Error at fs.js:729

okio.NodeJsFileSystemTest.fileHandleResizeLarger FAILED
    Error at fs.js:729
    ```

File: okio-nodefilesystem/src/jsMain/kotlin/okio/NodeJsFileSystem.kt [hunk 2]
```diff
@@ -87,35 +87,49 @@ object NodeJsFileSystem : FileSystem() {
...
+      try {
+        openFd(file, "r+")
+      } catch (e: FileNotFoundException) {
+        openFd(file, "wx+")
+      }
+    } else {
+      // On other platforms 'a+' does everything that we need.
+      openFd(file, "a+")
+    }
+    return NodeJsFileHandle(fd, readWrite = true)
```
[the remark this answers is not in the document]
Comment (oldergod) [reply okio-nodefilesystem/src/jsMain/kotlin/okio/NodeJsFileSystem.kt:112]: Tough

**Review by oldergod — APPROVED** [review]

**Pull request merged**